src/components/App/client-api.ts

```diff
@@ -0,0 +1,74 @@
+import App from './index';
+
+import { expose } from 'comlink';
+
+const API_VERSION = 1;
+
+export function exposeAPI(app: App) {
+  self.parent.postMessage({ type: 'READY', version: API_VERSION }, '*');
+  self.addEventListener('message', (ev: MessageEvent) => {
+    if (ev.data !== 'READY?') {
+      return;
+    }
+    ev.stopPropagation();
+    self.parent.postMessage({ type: 'READY', version: API_VERSION }, '*');
+  });
+  expose(new API(app), self.parent);
+}
+
+class API {
+  constructor(private app: App) { }
+
+  async setFile(blob: Blob, name: string) {
```

Comment:
Why not `async setFile(file: File | Fileish)` then?

Reply:
`Fileish` is our own type that we can’t really expose to a client and `File` alone suffers from the Edge bug, no?

Reply:
Ah, so it is because of the file issues in Edge.

I'm torn on this. Once we drop EdgeHTML support, we can drop `Fileish`, but with this API we're kinda locking ourselves into a weird pattern because of EdgeHTML.